### PR TITLE
T32244 discovery empty state

### PR DESCRIPTION
**Pull request by danigm**
https://phabricator.endlessm.com/T32244

### CODE REVIEW

File: kolibri_explore_plugin/assets/src/constants.js [hunk 1]
```diff
@@ -26,3 +26,5 @@ export const pageNameToModuleMap = {
 export const CUSTOM_PRESENTATION_TITLE = 'custom-channel-ui';
 
 export const CarouselItems = 5;
+
+export const searchTerms = ['STEM', 'Games', 'Fitness', 'Cooking', 'Arts'];
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/constants.js:30]: Yes, thanks! This should be a constant from day 1.

File: kolibri_explore_plugin/assets/src/views/SearchPage.vue [hunk 2]
```diff
@@ -101,25 +101,13 @@
...
-        const channels = [];
-        const allChannels = [...this.channels];
-        while (channels.length < this.columns && allChannels.length > 0) {
-          const index = _.random(0, allChannels.length - 1);
-          const [channel] = allChannels.splice(index, 1);
-          channels.push(channel);
-        }
-        return [channels];
+      isLoading() {
+        return this.loading && !!this.query.trim();
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/views/SearchPage.vue:110]: :+1:

File: kolibri_explore_plugin/assets/src/views/SearchPage.vue [hunk 3]
```diff
@@ -223,7 +217,7 @@
   @import '../styles';
 
   .main {
-    background-color: white;
+    background-color: $gray-300;
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/views/SearchPage.vue:220]: :+1:

File: kolibri_explore_plugin/assets/src/views/SearchPage.vue [hunk 4]
```diff
@@ -59,18 +67,7 @@
...
-        <h4 class="text-muted">
-          Featured Channels
-        </h4>
-        <ChannelCardGroup
-          :rows="recommended"
-          :columns="columns"
-          @card-click="goToChannel"
-        />
-      </b-container>
-    </div>
```
Comment (manuq) [inline kolibri_explore_plugin/assets/src/views/SearchPage.vue:73]: I see recommended channels in the specs, even for the minimal version:
Captura de pantalla de 2021-07-12 10-55-21

So I would keep this.

**Review by manuq — CHANGES_REQUESTED** [review]
The recommended channels should be kept according to specs. And there should be "some" space between buttons and the row of channels. This is a bit hidden in this comment: https://phabricator.endlessm.com/T32161#898155

The rest looks good.

File: kolibri_explore_plugin/assets/src/views/SearchPage.vue (hunk 4 above)
Comment (danigm) [reply kolibri_explore_plugin/assets/src/views/SearchPage.vue]: This is for the "No results", that's a bit different from the "Empty state". In any case. I've recovered the recommended in the code and just show it when the result is empty, but not in the first view with an empty search query.

**Review by manuq — APPROVED** [review]

**Pull request merged**